### PR TITLE
feature(mobile): allow app to be used offline

### PR DESCRIPTION
**Pull request by fyfrey**
- [x] stop being sent back to login screen on network errors
- [x] load cached thumbnail if preview/original fails to load
- [x] hide logging spam from flutter network errors for loading images. Display Error Icons instead
- [x] allow to skip server login on cold start without network

### CODE REVIEW

File: mobile/lib/modules/search/views/search_result_page.dart [hunk 1]
```diff
@@ -119,7 +119,12 @@ class SearchResultPage extends HookConsumerWidget {
           settings.getSetting(AppSettingsEnum.storageIndicator);
 
       if (searchResultPageState.isError) {
-        return const Text("Error");
+        return const Padding(
+          padding: EdgeInsets.all(12),
+          child: Text(
+            "Error.\nPlease check your network connection, make sure the server is reachable and app/server versions are compatible.",
```
Comment (michelheusschen) [inline mobile/lib/modules/search/views/search_result_page.dart:125]: Can this be made translatable?

Comment (zoodyy) [reply mobile/lib/modules/search/views/search_result_page.dart]: done

**Comment by alextran1502** [comment]
Hi Fynn, thank you for the PR. I just completed my round of functional testing, and I have a few questions.

First, my offline mode means turning off WIFI/Cellular/BLE and putting the phone in airplane mode.

1. In offline mode, any thumbnails that have been cached will display, but you cannot click on them to view, correct?
2. If an asset is on the device offline, the thumbnail should show up on the timeline, correct? I have some assets that got shown and some that didn't.
3. When in offline mode, when I click on the button to access the backup controller page, it takes a while (probably until HTTP request timeout) then it would navigate. Should we expect it to navigate as soon as we click on the button?

**Comment by zoodyy** [comment]
Thanks for testing this, Alex!

> First, my offline mode means turning off WIFI/Cellular/BLE and putting the phone in airplane mode.

Okay. It also applies if a) network is connected but not working (for whatever reason) or b) the API server is down.

>     1. In offline mode, any thumbnails that have been cached will display, but you cannot click on them to view, correct?

Thumbnails will display if a) the asset is stored locally on device or b) it has previously been cached. You can click on them to view. For local assets, the original is loaded as always. For remote assets, the preview/original is loaded (and shown if cached), else the cached thumbnail is shown as a fallback. 

>     2. If an asset is on the device offline, the thumbnail should show up on the timeline, correct? I have some assets that got shown and some that didn't.

Correct. Assets on device (local) are always shown both thumb and original (if in slide/detail view)

>     3. When in offline mode, when I click on the button to access the backup controller page, it takes a while (probably until HTTP request timeout) then it will navigate. Should we expect it to navigate as soon as we click on the button?

I also noticed some slight delay. Not sure why the page is not loading immediately. It should at least display the default state of all values.. I guess this is a riverpod issue or our `build` method in `BackupControllerPage` does something wrong 

>     4. In offline mode, it hangs on the splash screen when I kill the app and reopen it. I believe it is trying to wait for the response of the login process as well.
Hangs forever? For me, it takes a while longer (waiting for a server response) but then opens the app after a short (maybe 1 second) delay.

Not sure what's the best approach to speed up waiting on server responses.

**Comment by michelheusschen** [comment]
> Hangs forever? For me, it takes a while longer (waiting for a server response) but then opens the app after a short (maybe 1 second) delay.
> 
> Not sure what's the best approach to speed up waiting on server responses.

Maybe don't wait at all and start in offline mode by default. Then loading can happen in the background, but if it takes too long the user is made aware of this. Either by a popup or message on a page when a connection is necessary or an offline icon somewhere. The app could also occasionally check for a network connection and switch between online/offline mode.

Just my thoughts on an ideal solution. Depending on the current implementation, this could be quite some work to add but maybe some things are more straightforward. If it's too much, it can always be done in a future PR.

**Comment by zoodyy** [comment]
> Just my thoughts on an ideal solution. Depending on the current implementation, this could be quite some work to add but maybe some things are more straightforward. If it's too much, it can always be done in a future PR.

Yeah, that is out of scope for this PR. Would need quite a large refactor because everything is historically server-centric in the app. Better suited for a follow-up PR.

**Comment by martyfuhry** [comment]
Works well for me! I tried it offline & online, and gallery photos and videos loaded nicely. I also launched it while it airplane mode and it brought me right to the main timeline. Didn't notice if it took any longer.

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**